torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -509,7 +529,7 @@ void ProcessGroupAgent::pollTimedOutRPCs() {
       futureTimeoutCV_.wait_for(lock, sleepTime);
     }
 
-    if (shutdown_.load()) {
+    if (!rpcRunning_.load()) {
```

Comment:
This check is not strictly necessary, but can help performance. If the `shutdown` thread wakes up the CV here, then `rpcRunning` will always be false, without this check, we would continue to `processTimedOutFutures`, which would be a no-op.